src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -53,42 +62,65 @@ class ProtobufPlugin implements Plugin<Project> {
+      // has been applied then we will assume that none of prerequisite plugins were specified and we will
+      // throw an Exception to alert the user of this configuration issue.
+      Action<? super AppliedPlugin> applyWithPrerequisitePlugin = { prerequisitePlugin ->
+        if (wasApplied) {
+          project.logger.warn('The com.google.protobuf plugin was already applied to the project: ' + project.path
+              + ' and will not be applied again after plugin: ' + prerequisitePlugin.id)
+
+        } else {
+          wasApplied = true
+
+          doApply()
         }
+      }
 
-        if (!project.plugins.hasPlugin('java') && !Utils.isAndroidProject(project)) {
-            throw new GradleException('Please apply the Java plugin or the Android plugin first.'
-                + ' This error may also be a result of using the new Gradle plugins DSL.'
-                + ' Please use the traditional "apply" function as described in README.md.')
+      prerequisitePluginOptions.each { pluginName ->
+        project.pluginManager.withPlugin(pluginName, applyWithPrerequisitePlugin)
+      }
+
+      project.afterEvaluate {
+        if (!wasApplied) {
+          throw new GradleException('The com.google.protobuf plugin could not be applied during project evaluation.'
+              + ' The Java plugin or one of the Android plugins must be applied to the project first.')
         }
+      }
+    }
 
-        // Provides the osdetector extension
-        project.apply plugin: 'osdetector'
+    private void doApply() {
+      // Provides the osdetector extension
+      project.apply plugin: 'osdetector'
 
-        project.convention.plugins.protobuf = new ProtobufConvention(project, fileResolver);
+      project.convention.plugins.protobuf = new ProtobufConvention(project, fileResolver);
 
-        addSourceSetExtensions()
-        getSourceSets().all { sourceSet ->
-          createConfiguration(sourceSet.name)
-        }
-        project.afterEvaluate {
-          // The Android variants are only available at this point.
-          addProtoTasks()
-          project.protobuf.runTaskConfigClosures()
-          // Disallow user configuration outside the config closures, because
-          // next in linkGenerateProtoTasksToJavaCompile() we add generated,
-          // outputs to the inputs of javaCompile tasks, and any new codegen
-          // plugin output added after this point won't be added to javaCompile
-          // tasks.
-          project.protobuf.generateProtoTasks.all()*.doneConfig()
-          linkGenerateProtoTasksToJavaCompile()
-          // protoc and codegen plugin configuration may change through the protobuf{}
-          // block. Only at this point the configuration has been finalized.
-          project.protobuf.tools.resolve()
-        }
+      addSourceSetExtensions()
+      getSourceSets().all { sourceSet ->
+        createConfiguration(sourceSet.name)
+      }
+
```

Comment:
The following content should stay in `project.afterEvaluate`. Otherwise they will be run as soon as the prerequisite applies, at which point the `protobuf` DSL may have not been evaluated yet.